content/post/performance-metrics-overview.md

```diff
@@ -0,0 +1,111 @@
++++
+
+Metrics play an integral part in providing confidence in a high-performance software system. Whether you’re dealing with a data processing framework or a web server, metrics provide insight into whether your system is performing as expected. A direct impact of capturing metrics is the performance cost to the system they're for.
+
+When we first started development on [Wallaroo](https://github.com/WallarooLabs/Wallaroo), our high-throughput, low-latency, and elastic data processing framework, we were aiming to meet growing demand in the data processing landscape. Companies are relying on more and more data all while expecting to process that data as quickly as possible for, among other things, faster, better decision making, and cost reduction. Because of this, we set out to develop Wallaroo with three core principles in mind:
+
+1. High-Throughput
+2. Low-Latency
+3. 1/3 of the infrastructure costs/needs compared to existing solutions.
+
+Proving to our users that we were delivering on these principles was an essential part of the early development cycle of Wallaroo. How could we prove we were delivering on these principles? We needed to capture metrics. Metrics are an excellent way to spot bottlenecks in your system. Providing meaningful metrics to our users would mean they could make quicker iterations in their development cycle by being able to spot specific bottlenecks in various locations of our system. An added benefit that came from capturing these metrics was that it also sped up the Wallaroo development cycle as we were able to use them as guidelines to [measure the “cost” of every feature](https://blog.wallaroolabs.com/2017/06/whats-the-secret-sauce/#measure-the-cost-of-every-feature).
+
+In the end, what came of our metrics capturing process became a web UI that provides Wallaroo users introspection for various parts of the applications they develop. Here's what the result looks like:
+
+![application dashboard page](/images/post/performance-metrics-overview/application-dashboard-page.png)
+
+In this post, I'll cover some of the design decisions we made regarding our metrics capturing system to maintain the high performance our system was promising.
+
+## Metrics Aren't Free
+
+Adding introspection via metrics to your system inherently adds an overhead. However, the impact of that overhead is determined by how you capture metrics and what information you choose to capture. Not only is there a ton of information to capture, but there are also many different ways to capture it. The biggest problem we face while determining how we capture metrics is maintaining a low overhead. A computationally heavy metrics capturing system could mean lower throughput, higher latencies, and costlier infrastructure, all things we want to avoid. When approaching this type of problem, you need to determine how much information you need to capture to provide meaningful details via your metrics while minimizing the performance impact on your system.
```

Comment:
It seems

> The biggest problem we face while determining how we capture metrics is maintaining a low overhead.

should be

> The biggest problem **you** face while determining how **you** capture metrics is maintaining a low overhead.

in order to keep with the `you` based point of view in the rest of the paragraph.

Similarly

> A computationally heavy metrics capturing system could mean lower throughput, higher latencies, and costlier infrastructure, all things we want to avoid.

should be

> A computationally heavy metrics capturing system could mean lower throughput, higher latencies, and costlier infrastructure, all things **you** want to avoid.